.clang-tidy

```diff
@@ -6,23 +6,16 @@ Checks: '
          -clang-analyzer-security.insecureAPI.rand,
          -readability-implicit-bool-conversion,
          -cppcoreguidelines-pro-type-reinterpret-cast,
-         -readability-misleading-indentation,
          -modernize-pass-by-value,
          -cppcoreguidelines-pro-bounds-constant-array-index,
          -cppcoreguidelines-pro-type-static-cast-downcast,
```

Comment:
Add a comment that we ensure success with type enums?

Reply:
You mean something like `static_cast<ColumnType>` which we obtain via `data_type()`? Or something else?

Reply:
Yes, or `static_cast<JoinNode&>(*node)` after resolving `node->type`, `static_cast<LikeExpression&>(*expression)` after getting `expression-type`, ...

Reply:
What about `#   - cppcoreguidelines-pro-type-static-cast-downcast           We use static downcasts when we can safely obtain the type (e.g., static casts of LQPNodes via node->type)`?